### DISCUSSION TITLE
Possible to add more than one related persons in one query?

### DISCUSSION THREAD

**Discussion by yongkansage** [Gremlinq/ExRam.Gremlinq#308] [Q&A]
When there is 1 to 1 related person, a single query can be written as example shown below to create nodes and edge together.

   // Add Persons and and edge in between in one single query!
    await _g
        .AddV(new Person { Name = "Bob", Age = 36 })
        .AddE<Knows>()
        .To(__ => __
            .AddV(new Person { Name = "Jeff", Age = 27 }))
        .FirstAsync();

Is it possible to write a single query to create nodes and edges together for 1 to many related case, such as Bob knows Jeff, Daniel and Tom?

**Comment by danielcweber** [comment]
Try doing it in reverse and nesting the queries, each time returning the OutV (which is Bob), like below. For a general introduction on Gremlin, check out the [Tinkerpop docs](https://tinkerpop.apache.org/docs/current/reference/). For all kinds of general questions about the Gremlin query language, check out the [Gremlin users mailing list](https://groups.google.com/g/gremlin-users).

```
await g
    .AddV(new Person { Name = "Jeff" })
    .AddE()
    .From(_ => _
        .AddV(new Person { Name = "Daniel" })
        .AddE()
        .From(_ => _
             .AddV(new Person { Name = "Tom" })
             .AddE()
             .From(_ => _
                   .AddV(new Person { Name = "Bob" })
            .OutV())
        .OutV())
    .FirstAsync();

**Reply by yongkansage** [reply]
Thanks! If the list of persons Bob knows is not a fixed size of collection, what would be your advice to construct the query?

**Reply by danielcweber** [reply]
Recursive code I guess...

**Reply by yongkansage** [reply]
This is the pattern I ended up after some experimentation:

            var v = g.AddV<Root>(root)
                        .AddE<LinkedTo>()
                            .To(_ => _
                                .AddV<Leaf>(leaf1))
                            .OutV<Root>()
                            .AddE<LinkedTo>()
                                .To(_ => _
                                    .AddV<Leaf>(leaf2))
                                .OutV<Root>()
                                .AddE<LinkedTo>()
                                    .To(_ => _
                                        .AddV<Leaf>(leaf3)
                                    .OutV<Root>();
